src/crepe/system/RenderSystem.cpp

```diff
@@ -24,21 +26,59 @@ void RenderSystem::update_camera() {
 		this->curr_cam = &cam;
 	}
 }
-void RenderSystem::render_sprites() const {
+
+bool RenderSystem::render_particle(const Sprite & sprite, Transform tm) {
+
 	ComponentManager & mgr = this->component_manager;
+	SDLContext & render = SDLContext::get_instance();
+
+	auto emitters = mgr.get_components_by_id<ParticleEmitter>(sprite.game_object_id);
+
+	bool rendering_particles = false;
 
-	std::vector<std::reference_wrapper<Sprite>> sprites = mgr.get_components_by_type<Sprite>();
+	for (const ParticleEmitter & em : emitters) {
+		if (!em.active) continue;
+		if (!(em.data.sprite.game_object_id == sprite.game_object_id)) continue;
 
+		rendering_particles = true;
+
+		for (const Particle & p : em.data.particles) {
+			if (!p.active) continue;
+			tm.position = p.position;
+			tm.rotation = p.angle;
+			render.draw(em.data.sprite, tm, *curr_cam);
```

Comment:
Dat zou `*this->curr_cam` moeten zijn, maar als je `master` merget kun je van `curr_cam` gelijk beter een `OptionalRef<Camera>` maken. Dit komt vaker voor dus check het overal.